src/solver/conv_hip_implicit_gemm_fwd_xdlops.cpp

```diff
@@ -227,6 +226,40 @@ void RunCKSolution(const Handle& handle,
     }
 }
 
+InvokerFactory MakeInvokerFactoryHipImplGemmFwdXdlops(const ProblemDescription& problem, 
+    const PerformanceConfigHipImplicitGemmFwdXdlops& config) {
+
+  auto ck_args = CKArgs{problem};
+  miopenDataType_t data_type = problem.conv_problem.GetInDataType();
+  auto config_idx = config.index;
+
+
+  InvokerFactory ret = [ck_args, data_type, config_idx] (const std::vector<Kernel>& kernels) {
+        std::ignore = kernels;
+        return [ck_args, data_type, config_idx] (const Handle& handle, const AnyInvokeParams& primitive_parameters) {
+            switch(data_type)
```

Comment:
[R] Overhead of switch is small, but I recommend moving it into factory anyway. That would make the final Invoker object smaller at least.

Reply:
I don't think it would be worth the effort. There would be a whole bunch of copy-paste lambdas for each case with one line difference.

Reply:
...or better to the solver which can MIOPEN_LOG_E and then return error code in the solution for the unhandled cases (instead of MIOPEN_THROW in the factory)

@DrizztDoUrden what is your opinion?

Reply:
Removed [R] ("recommendation") tag as this change is necessary to compute `conv_ptr` (https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2305#discussion_r1290087188)

Reply:
`switch` moved out in 9034f3e but `conv_ptr` and `invoker_ptr` cannot be moved out due to them being `non-copyable`. See the comment above (https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2305#discussion_r1294994063)

Reply:
[Resolved]